src/main/java/billgates/view/gui/TopMenuBar.java

```diff
@@ -34,69 +36,47 @@ private void initMenu() {
         this.add(Box.createRigidArea(new Dimension(ActionPanel.VERTICAL_GAP, 0)));
 
         this.add(this.importMenu);
+        this.add(Box.createRigidArea(new Dimension(ActionPanel.VERTICAL_GAP, 0)));
         this.add(this.settingsMenu);
-
-        // importMenu should be disabled  before signing in
-//        this.importMenu.setEnabled(false);
+        this.add(Box.createRigidArea(new Dimension(ActionPanel.VERTICAL_GAP, 0)));
+        this.add(this.helpMenu);
 
         // Add function to import menu
-        importMenu.addMenuListener(new MenuListener() {
+        importMenu.addMenuListener(new MenuAdaptor(){
             @Override
             public void menuSelected(MenuEvent e) {
                 importBills();
             }
-
-            @Override
-            public void menuDeselected(MenuEvent e) {
-
-            }
-
-            @Override
-            public void menuCanceled(MenuEvent e) {
-
-            }
         });
         this.add(Box.createRigidArea(new Dimension(ActionPanel.VERTICAL_GAP, 0)));
 
         // Add function to setting menu
-        settingsMenu.addMenuListener(new MenuListener() {
+        settingsMenu.addMenuListener(new MenuAdaptor() {
             @Override
             public void menuSelected(MenuEvent e) {
                 System.out.println("yes");
                 int[] setting = setting();
 
-                MainFrame.getBillPanel().changeColor(colors[setting[0]]);
-                MainFrame.getActionPanel().changeColor(colors[setting[0]]);
+                mainFrame.getBillPanel().changeColor(colors[setting[0]]);
+                mainFrame.getActionPanel().changeColor(colors[setting[0]]);
 
-                MainFrame.getActionPanel().changeFont(fonts[setting[1]]);
-                MainFrame.getBillPanel().changeFont(fonts[setting[1]]);
-
-            }
-
-            @Override
-            public void menuDeselected(MenuEvent e) {
-
-            }
-
-            @Override
-            public void menuCanceled(MenuEvent e) {
+                mainFrame.getActionPanel().changeFont(fonts[setting[1]]);
+                mainFrame.getBillPanel().changeFont(fonts[setting[1]]);
 
             }
         });
-
-
-        this.add(Box.createRigidArea(new Dimension(ActionPanel.VERTICAL_GAP, 0)));
-
-        this.add(this.helpMenu);
     }
 
     // Function to set up the setting dialog
     private int[] setting() {
-        SettingDialog setdlg = new SettingDialog(this);
+//        SettingDialog setdlg = mainFrame.getSettingDialog();
+        SettingDialog setdlg = new SettingDialog(mainFrame);
         int[] setting = new int[2];
         if (setdlg.exec()) {
             int colorTargetIndex = setdlg.getMyColor();
             int fontTargetIndex = setdlg.getMyFont();
+            SettingDialog.setFontField(fontTargetIndex);
+            SettingDialog.setColorField(colorTargetIndex);
```

Comment:
Seems like you have some problems with the action logic here.
These two lines shouldn't be there